tests/files/FileproxyTest.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+import os
+
+from coalib.files.Fileproxy import Fileproxy
+
+
+class FileproxyTest(unittest.TestCase):
+
+    def setUp(self):
+        fileproxy_test_dir = os.path.join(os.path.split(__file__)[0],
+                                          'FileproxyTestFiles')
+
+        self.testfile = os.path.join(fileproxy_test_dir, 'testfile.txt')
+        self.fileproxyobject = Fileproxy(self.testfile)
+
+    def test_str(self):
+        self.assertEqual(str(self.fileproxyobject),
+                         open(self.testfile).read())
+
+    def test_equal(self):
+        lines = open(self.testfile).readlines()
+        with open(self.testfile, 'a+') as file:
+            file.write("New line")
+
+        self.assertEqual(self.fileproxyobject, Fileproxy(self.testfile))
+
+        with open(self.testfile, 'w') as file:
+            file.writelines(lines)
+
+    def test_iter(self):
+        with open(self.testfile, 'r') as file:
+            self.assertEqual(tuple(self.fileproxyobject),
+                             tuple(file.readlines()))
+
+    def test_hashing(self):
+        lines = open(self.testfile).readlines()
```

Comment:
I dislike this way of testing, I'm not sure this test is even needed. If you think it is, make an own testcase object and make the file write back in a teardown method (read in setup)